include/options_netcdf.hxx

```diff
@@ -1,4 +0,0 @@
-#pragma once
-// BOUT++ header shim
-#warning Header "options_netcdf.hxx" has moved to "bout/options_netcdf.hxx". Run `bin/bout-v5-header-upgrader.py` to fix
```

Comment:
I think we should either remove all compat headers, or none.

I would be in favor of deleting, as we probably have by now anyway breaking changes that need adoption ...